alf/algorithms/dynamic_action_repeat_agent.py

```diff
@@ -188,14 +202,36 @@ def _generate_new_action(time_step, state):
 
     def rollout_step(self, time_step: TimeStep, state: ActionRepeatState):
         switch_action = self._should_switch_action(time_step, state)
+
+        # state.k is the current step index over K steps
         state = state._replace(
-            rl_reward=state.rl_reward + state.rl_discount * time_step.reward,
-            rl_discount=state.rl_discount * time_step.discount * self._gamma)
+            rl_reward=state.rl_reward + torch.pow(
```

Comment:
`rl_reward=state.rl_reward + state.rl_discount * time_step.reward`

Reply:
This will not necessarily be correct if time_step.discount is between 0 and 1 because rl_reward (t+1) already has one extra multiplied discount for training the previous time step (t).

Reply:
If time_step.discount is betwee 0 and 1, I think this place will be somewhat inconsistent with line 342 because there acc_mean only considers gamma^k.